combat-trainer.lic

```diff
@@ -2730,6 +2749,83 @@ class TrainerProcess
          'dumping some dirt', 'What were you referring')
   end
 
+  def pickbox(game_state, number='first')
+    DRC.message("@pet_count is #{@pet_count}.") if $debug_mode_ct
+    return unless @pet_box_source && @pet_count > 0
+    return if game_state.loaded || game_state.offhand?
+    return if left_hand && !game_state.currently_whirlwinding
+
+    if right_hand
+      if game_state.weapon_is_summoned?
+        echo 'combat-trainer::pickbox: weapon summoned' if $debug_mode_ct
+        if DRStats.warrior_mage?
+          bput("lower #{game_state.weapon_name} to ground", 'You lower')
+        else
+          bput("wear my #{game_state.weapon_name}", 'telekinetic force')
+        end
+      else
+        echo 'combat-trainer::pickbox: weapon NOT summoned' if $debug_mode_ct
+        @equipment_manager.stow_weapon(game_state.weapon_name)
+      end
+      had_weapon = true
+    end
+    game_state.sheath_whirlwind_offhand if game_state.currently_whirlwinding
+
+    num = 1
+    picked = false
+    while !picked
+      box = "box"
+      if !DRCI.in_hands?(box) && !DRCI.get_item?("#{number} #{box}", @pet_box_source)
+        box = "pouch"
+        DRCI.get_item?("#{number} #{box}", @pet_box_source) unless DRCI.in_hands?(box)
+      end
+      case bput("pick my #{box}", 'not making any progress', 'it opens.', "isn't locked", 'The lock feels warm', 'The lock looks weak, and rather than risk breaking it', /It's not even locked, why bother?/)
+      when /it opens/
+        waitrt?
+        bput("lock my #{box}", 'You quickly lock', 'already locked')
+        picked = true
+      when /isn't locked/
+        bput("close my #{box}", "You close", "That is already closed")
+        bput("lock my #{box}", "You quickly lock", "already locked")
+        picked = false
+      when /The lock feels warm/
+        num += 1
+        DRCI.put_away_item?(box, @pet_box_source)
+        @pet_count -= 1
+        if num == 2
+          ord = 'second'
+        elsif num == 3
+          ord = 'third'
+        elsif num = 4
+          ord = 'fourth'
+        end
+        return pickbox(game_state, number=ord)
+      when /The lock looks weak, and rather than risk breaking it/
+        DRCI.dispose_trash(box, @worn_trashcan, @worn_trashcan_verb)
+        @pet_count -= 1
+        pick = false
+      when /It's not even locked, why bother?/
+        DRCI.put_away_item?(box)
+        @pet_count -= 1
+        pick = false
+      end
+      break if @pet_count == 0
+    end
+    DRC.message("@pet_count is #{@pet_count}.") if $debug_mode_ct
+    waitrt?
+    DRCI.put_away_item?(box, @pet_box_source)
+
+    if had_weapon
```

Comment:
I wonder if `check_item_recovery` or `recover_item` would help here?

Reply:
`common-items` would need one additional `@@get_item_success_patterns` added for the moon blade, `from the air`, to use `recover_item`. I am going to test it or get someone to test it for MM if mine does not have the moon blade yet.